lighthouse-core/config/config-plugin.js

```diff
@@ -93,10 +93,14 @@ class ConfigPlugin {
       if (typeof weight !== 'number') {
         throw new Error(`${pluginName} has an invalid auditRef weight.`);
       }
+      if (typeof group !== 'string' && typeof group !== 'undefined') {
+        throw new Error(`${pluginName} has an invalid auditRef group.`);
+      }
 
       return {
         id,
         weight,
+        ...group && {group},
```

Comment:
I don't think I've seen this pattern before, definitely gave me pause.

This is all to avoid having `.group` be in the object but undefined, correct? Is there a case that this messes with to warrant the extra layer of mental parsing?

Reply:
You're right, I thought this was necessary but it isn't needed and is certainly unreadable. Removed.